x/storage/keeper/bucket_rate_limit.go

```diff
@@ -0,0 +1,310 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	paymenttypes "github.com/bnb-chain/greenfield/x/payment/types"
+	"github.com/bnb-chain/greenfield/x/storage/types"
+)
+
+func (k Keeper) SetBucketFlowRateLimit(ctx sdk.Context, operator sdk.AccAddress, bucketOwner sdk.AccAddress, paymentAccount sdk.AccAddress, bucketName string, rateLimit sdkmath.Int) error {
+	// check the operator is the same as the payment account owner
+	if !k.paymentKeeper.IsPaymentAccountOwner(ctx, paymentAccount, operator) {
+		return paymenttypes.ErrNotPaymentAccountOwner
+	}
+
+	// get the bucket
+	bucket, found := k.GetBucketInfo(ctx, bucketName)
+
+	// check the bucket owner is the same as the bucket owner of the bucket
+	if found && bucket.Owner != bucketOwner.String() {
+		return fmt.Errorf("the bucket owner is not the same as the bucket owner of the bucket")
```

Comment:
I think it is ok, dont have to check `bucket.Owner != bucketOwner.String()`, because we can not guarante this if the bucket is created later.